### PR TITLE
feat: Add eject and add expenses from platform

### PR DESCRIPTION
**Pull request by bistaastha**
Tests will be fixed in follow up PR

### CODE REVIEW

File: src/app/core/services/platform/v1/approver/reports.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  ejectExpenses(rptId: string, txnId: string, comment?: string[]): Observable<void> {
+    const payload = {
+      data: {
+        id: rptId,
+        expense_ids: [txnId],
+      },
+      reason: comment,
+    };
+    return this.approverPlatformApiService.post<void>('/reports/eject_expenses', payload);
```
Comment (bistaastha) [inline src/app/core/services/platform/v1/approver/reports.service.ts:19]: returning a void observable following the existing pattern for remove transaction API

File: src/app/core/services/platform/v1/approver/reports.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { TestBed } from '@angular/core/testing';
+import { ApproverReportsService } from './reports.service';
+import { ApproverPlatformApiService } from '../../../approver-platform-api.service';
+
+describe('ApproverReportsService', () => {
+  let reportsService: ApproverReportsService;
```
Comment (Chethan-Fyle) [inline src/app/core/services/platform/v1/approver/reports.service.spec.ts:6]: `reportsService` --> `approverReportsService`.

File: src/app/core/services/platform/v1/spender/reports.service.spec.ts [hunk 3]
```diff
@@ -13,19 +13,19 @@ import {
 } from 'src/app/core/mock-data/platform-report.data';
 import { ReportsQueryParams } from 'src/app/core/models/platform/v1/reports-query-params.model';
 
-describe('ReportsService', () => {
-  let reportsService: ReportsService;
+describe('SpenderReportsService', () => {
+  let reportsService: SpenderReportsService;
```
Comment (Chethan-Fyle) [inline src/app/core/services/platform/v1/spender/reports.service.spec.ts:17]: `reportsService` --> `spenderReportsService`.

File: src/app/fyle/view-expense/view-expense.page.ts [hunk 4]
```diff
@@ -150,7 +151,8 @@ export class ViewExpensePage {
     private categoriesService: CategoriesService,
     private dependentFieldsService: DependentFieldsService,
     private spenderExpensesService: SpenderExpensesService,
-    private approverExpensesService: ApproverExpensesService
+    private approverExpensesService: ApproverExpensesService,
+    private reportsService: ApproverReportsService
```
Comment (Chethan-Fyle) [inline src/app/fyle/view-expense/view-expense.page.ts:155]: `reportsService` --> `approverReportsService`

**Review by Chethan-Fyle — CHANGES_REQUESTED** [review]
Minor comments.

File: src/app/core/services/platform/v1/approver/reports.service.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { Observable } from 'rxjs';
+import { ApproverPlatformApiService } from '../../../approver-platform-api.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApproverReportsService {
+  constructor(private approverPlatformApiService: ApproverPlatformApiService) {}
+
+  ejectExpenses(rptId: string, txnId: string, comment?: string[]): Observable<void> {
```
Comment (mvaishnavi) [inline src/app/core/services/platform/v1/approver/reports.service.ts:11]: `expenseId`
let's fix this terminology for the add Expense to report service method too.

**Review by mvaishnavi — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**